src/cocotb/regression.py

```diff
@@ -90,24 +91,62 @@
         assert False, "pytest.raises doesn't raise an exception when it fails"
 
 
-class _Test:
-    _id_count = 0  # used by the RegressionManager to sort tests in definition order
+class Test:
+    """A cocotb test in a regression.
+
+    Args:
+        func:
+            The test function object.
+
+        name:
+            The name of the test function.
+            Defaults to ``func.__qualname__`` (the dotted path to the test function in the module)
+
+        module:
+            The name of the module containing the test function.
+            Defaults to ``func.__module__`` (the name of the module containing the test function).
+
+        doc:
+            The docstring for the test.
+            Defaults to ``func.__doc__`` (the docstring of the test function).
+
+        timeout_time:
+            Simulation time duration before timeout occurs.
+
+        timeout_unit:
+            Units of ``timeout_time``, accepts any units that :class:`~cocotb.triggers.Timer` does.
+
+        expect_fail:
+            If ``True`` and the test fails a functional check via an ``assert`` statement, :class:`pytest.raises`,
+            :class:`pytest.warns`, or :class:`pytest.deprecated_call`, the test is considered to have passed.
+            If ``True`` and the test passes successfully, the test is considered to have failed.
+
+        expect_error:
+            Mark the result as a pass only if one of the given exception types is raised in the test.
+
+        skip:
+            Don't execute this test as part of the regression.
+            The test can still be run manually by setting :make:var:`TESTCASE`.
```

Comment:
```suggestion
            The test can still be run manually by setting :envvar:`TESTCASE`.
```
(doesn't link otherwise, we have another such occurrence of `:make:var:` in the release notes.)